specification/dartLangSpec.tex

```diff
@@ -673,16 +674,34 @@ \subsection{Scoping}
 \LMLabel{scoping}
 
 \LMHash{}%
-A \Index{namespace} is a mapping from names to declarations.
-It may be considered as a partial function.
-It may also be seen as a set of bindings from a name to a declaration,
-$n\mapsto{}D$,
-with the constraint that if a namespace has both
-$n\mapsto{}D_1$ and $n\mapsto{}D_2$
-then $D_1$ and $D_2$ is the same declaration.
-When a namespace is seen as a set of mappings we may also refer
-to each binding as having a \NoIndex{key}
-which is mapped to a \NoIndex{value}.
+A \Index{namespace} is a partial function that maps names to namespace values.
+A \Index{name} is a lexical token which is an \synt{IDENTIFIER},
+an \synt{IDENTIFIER} followed by \lit{=}, or
+an \synt{operator};
```

Comment:
We don't store constructors under their constructor name then.
(Do we store constructors in namespaces at all? If we do, we need to either do something for `ClassName.name`, or if we only store `name`, then for the `ClassName` constructor - unless we store that under `ClassName` directly instead of "<empty>").

Also need to account for `unary-` with the operators. It's *names* of operators that we map from, not their syntax, and that it differs for `unary-`.